coalib/output/printers/LogPrinter.py

```diff
@@ -1,54 +1,135 @@
+import sys
 import traceback
+import logging
 
-from pyprint.ColorPrinter import ColorPrinter
+from pyprint.Printer import Printer
 
-from coalib.output.printers.LOG_LEVEL import LOG_LEVEL, LOG_LEVEL_COLORS
+from coalib.output.printers.LOG_LEVEL import LOG_LEVEL
 from coalib.processes.communication.LogMessage import LogMessage
 
 
+class PrinterHandler(logging.StreamHandler):
+
+    def __init__(self, printer=None, *args, **kwargs):
+        self.printer = printer
+        super(PrinterHandler, self).__init__(*args, **kwargs)
+
+    def emit(self, record):
+        msg = self.format(record)
+        if self.printer:
+            self.printer.print(msg)
+        else:
+            sys.stdout.write(msg + "\n")
+
+
 class LogPrinter:
     """
     The LogPrinter class allows to print log messages to an underlying Printer.
 
     This class is an adapter, means you can create a LogPrinter from every
     existing Printer instance.
     """
+    _loggers = {}
+
+    @staticmethod
+    def _get_logger(printer, logger_name, time_fmt=None):
+        """
+        Returns existing or newly created and configured logger.
+        :param logger_name: name of the logger to be returned.
+        :param printer:     printer to initialize logger with if it doesn't
+                            exist, yet.
```

Comment:
@tushar-rishav Yes, for now I'm more interested in general approach. :smile: What's your opinion?